nwbinspector/checks/tables.py

```diff
@@ -32,6 +32,14 @@ def check_empty_table(table: DynamicTable):
         return InspectorMessage(message="This table has no data added to it.")
 
 
+@register_check(importance=Importance.BEST_PRACTICE_VIOLATION, neurodata_type=DynamicTable)
+def check_column_data_is_not_none(table: DynamicTable, nelems: int = 200):
+    """Check column values in a DynamicTable to ensure they are not None."""
+    for column in table.columns:
+        if any((x is None for x in column.data[:nelems])):
+            yield InspectorMessage(message=f"The column '{column.name}' has 'data' set to None.")
```

Comment:
How would this even happen? There is no `None` type in HDF5.

Reply:
See above: Actually, similar to what I'm finding in PR https://github.com/NeurodataWithoutBorders/nwbinspector/pull/135, it doesn't seem possible to actually complete an io.write() with any None values anywhere in the table. Do we even need this kind of check, then?

> Actually, similar to what I'm finding in PR https://github.com/NeurodataWithoutBorders/nwbinspector/pull/135, it doesn't seem possible to actually complete an io.write() with any None values anywhere in the table. Do we even need this kind of check, then?

Reply:
it's more of an issue of how _to_ handle missing metadata

Reply:
In that case, perhaps a separate check "check_missing_column_data" with logic "if len(column.data) == 0:"?

Note we already have one higher level case of this for entirely empty tables (no rows/cols): https://github.com/NeurodataWithoutBorders/nwbinspector/blob/dev/nwbinspector/checks/tables.py#L28-L32